design/security/high_availability.md

```diff
@@ -0,0 +1,66 @@
+# High availability
+
+The Obscuro Sequencer must have HA capabilities. The reasoning is covered in the "Fast finality" design docs.
+
+The requirement is that the service must continue even if the enclave of the sequencer crashes and is corrupted.
+
+One option is to require a governance event to whitelist a new sequencer enclave in case the current one becomes unavailable.
+This is not a good solution, because it would lead to significant downtime.
+
+The preferred option is to mark multiple enclaves as sequencers, and make it the responsibility of the operator to switch
+between them to ensure that service goes on uninterrupted.
+
+## MEV Protection 
+
+The Sequencer is the only Obscuro node capable of producing Light Batches, and thus the only node that can in theory
+attempt to extract value via MEV. If the operator has multiple enclaves available it can both provide an uninterrupted service
+and extract some value at the same time.
+
+Introducing startup delays does not help too much in this case, because the operator could hold key transactions for longer.
+
+
+An alternative solution is to introduce restart transparency, such that the Obscuro network can assess the likelyhood
+of bad behaviour.  
+
+On a high level, all enclave restart events of all the sequencer whitelisted enclaves must be published to the network.
+
+### Protocol
+
+Each Batch will have three elements:
```

Comment:
Does the protocol payload really need to be part of each batch?

Reply:
good question. 
Initially, I had a question mark there but removed it before committing.

My thinking was that since it's a low-cost action (little computation and no storage cost), we might as well do it every time.

Reply:
If it never goes to the data availability yeah, it is low cost. If we do have to include it in rollups (as we include the full batches rn) it might become a bit too expensive (given that including the batch metadata is also not really cheap when doing it every 1 second)

Reply:
I don't think it needs to be in the rollup. 
It's not consensus stuff. It's just a log of activity.